File: ui-core/src/types/family.ts
import {
  Enrollee,
  EnrolleeRelation
} from 'src/types/user'

export type Family = {
    id: string
    createdAt: number
    lastUpdatedAt: number
    probandEnrolleeId: string
    shortcode: string
    studyEnvironmentId: string
    members?: Enrollee[]
    proband?: Enrollee
    relations?: EnrolleeRelation[]
}

export type FamilyEnrollee = {
    familyId: string
    enrolleeId: string
    createdAt: number
}

File: core/src/main/java/bio/terra/pearl/core/model/participant/Family.java
package bio.terra.pearl.core.model.participant;

import bio.terra.pearl.core.model.BaseEntity;
import bio.terra.pearl.core.model.study.StudyEnvAttached;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.List;
import java.util.UUID;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
public class Family extends BaseEntity implements StudyEnvAttached {
    private UUID probandEnrolleeId;
    private UUID studyEnvironmentId;
    private String shortcode;

    // optional related entities
    private Enrollee proband;
    private List<Enrollee> members;
    private List<EnrolleeRelation> relations;
}
